contracts/contracts/strategies/Generalized4626Strategy.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title OETH Generalized 4626 Strategy
+ * @notice Investment strategy for vaults supporting ERC4626
+ * @author Origin Protocol Inc
+ */
+import { IERC4626 } from "../../lib/openzeppelin/interfaces/IERC4626.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20, InitializableAbstractStrategy } from "../utils/InitializableAbstractStrategy.sol";
+
+contract Generalized4626Strategy is InitializableAbstractStrategy {
+    using SafeERC20 for IERC20;
+
+    IERC20 shareToken;
+    IERC20 assetToken;
+
+    /**
+     * @dev Deposit assets by converting them to shares
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        _deposit(_asset, _amount);
+    }
+
+    /**
+     * @dev Deposit assets by converting them to shares
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     */
+    function _deposit(address _asset, uint256 _amount) internal {
+        require(_amount > 0, "Must deposit something");
+        require(_asset == address(assetToken), "Unexpected asset address");
+
+        IERC4626(platformAddress).deposit(_amount, address(this));
+        emit Deposit(_asset, address(shareToken), _amount);
+    }
+
+    /**
+     * @dev Deposit the entire balance of assetToken to gain shareToken
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 balance = assetToken.balanceOf(address(this));
+        if (balance > 0) {
+            _deposit(address(assetToken), balance);
+        }
+    }
+
+    /**
+     * @dev Withdraw asset by burning shares
+     * @param _recipient Address to receive withdrawn asset
+     * @param _asset Address of asset to withdraw
+     * @param _amount Amount of asset to withdraw
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient != address(0), "Must specify recipient");
+        require(_asset == address(assetToken), "Unexpected asset address");
+
+        IERC4626(platformAddress).withdraw(_amount, _recipient, address(this));
```

Comment:
Perhaps we should check that the return value from calling `withdraw` here matches the _amount.

Reply:
Slither returns amount of shares that were burned (sfrxETH tokens) in this case and not the amount of assets that have been withdrawn (frxETH token). 

From the [source code](https://github.com/FraxFinance/frxETH-public/blob/7f7731dbc93154131aba6e741b6116da05b25662/flattened.sol#L688) it does seem that the correct amount of `assets` gets transferred. Though this applies only for `sfrxToken` contract of course. 

We could run `shares = previewWithdraw(assets)` with the withdrawing assets and confirm that the `shares` amount match between what `previewWithdraw` returns and what `withdrawal` returns.

Reply:
Probably good just ignoring then